### PR TITLE
Fixes MYRIAD-114

### PR DESCRIPTION
**Pull request by klucar**
Fixed gulp build to not include local paths and gradle-ized build.
Removed public/ from git repo and added it to .gitignore.

### CODE REVIEW

File: myriad-scheduler/build.gradle [hunk 1]
```diff
@@ -69,3 +74,36 @@ sourceSets {
...
+    // Set the work directory where node_modules should be located
+    nodeModulesDir = file("${project.projectDir}/src/main/resources/webapp")
+}
+
+gulp {
+    // Set the directory where gulpfile.js should be found
+    workDir = file("${project.projectDir}/src/main/resources/webapp")
+
+    // Whether colors should output on the terminal
+    colors = false
```
Comment (smarella) [inline myriad-scheduler/build.gradle]: No colors?

**Comment by smarella** [comment]
Tested it on my dev setup, but the RM can't come up. It throws me the following null pointer:

```
Caused by: java.lang.NullPointerException
        at org.apache.myriad.webapp.MyriadWebServer.start(MyriadWebServer.java:64)
        at org.apache.myriad.Main.initWebApp(Main.java:119)
        at org.apache.myriad.Main.run(Main.java:91)
        at org.apache.myriad.Main.initialize(Main.java:76)
        at org.apache.myriad.scheduler.yarn.interceptor.MyriadInitializationInterceptor.init(MyriadInitializationInterceptor.java:51)
        ... 16 more
```
The corresponding code in the webserver is:
```java
    String staticDir = this.getClass().getClassLoader().getResource("webapp/public").toExternalForm();
```
The myriad-scheduler-0.1.jar doesn't have "webapp/public". The build doesn't seem to copy "webapp/public" under the build directory.

**Comment by klucar** [comment]
I saw that once but couldn't reproduce it. I figured it was just from me
trying to get everything working. A few questions: did you gradle or
gradlew? Were the files generated in src/main/resources/webapp/public and
just didn't make it into the jar? I'll try to reproduce again this morning.

File: myriad-scheduler/build.gradle (hunk 1 above)
Comment (klucar) [inline myriad-scheduler/build.gradle]: ha! sometimes colors just print weird control characters if your terminal can't handle it. you're not missing anything too exciting.

**Comment by smarella** [comment]
I did "gradlew".

Yes. The files were generated under src/main/resources/webapp/public, but
didn't get copied over to myriad-scheduler/build/resources/main/webapp and
subsequently didn't make it into the myriad-scheduler-0.1.jar.

Santosh

**Comment by klucar** [comment]
I updated the PR. I moved the web app build to before process resources.
Brandon and I both tried it and saw the files in the jar now.

**Comment by smarella** [comment]
awesome. I'll test it out again before merging.

Santosh

**Comment by smarella** [comment]
Tested it out. The build is good, RM comes up alright. But the UI doesn't seem to render. Only the favicon and the page title seem to appear.

**Comment by klucar** [comment]
I found an error in the shutdown component that kept it from rendering. I actually think I want to re-do the interface to make it more like the website design, but that can wait a bit.

**Comment by smarella** [comment]
@klucar - any chance this could be fixed today? This is a blocker for 0.1.0.

**Comment by klucar** [comment]
I thought I fixed it, maybe I didn't push. It was a simple error in the
JavaScript. Check the PR. I'm on mobile now but can look at it later.

**Comment by smarella** [comment]
@klucar - you are right. I see a commit added to the PR. Not sure why I don't receive the git notifications, though. I'll test this out locally and merge in if the UI renders correctly. Thanks!

**Pull request closed without merging**